src/ast/semantic_analyser.cpp

```diff
@@ -76,6 +76,14 @@ void SemanticAnalyser::visit(PositionalParameter &param)
 
 void SemanticAnalyser::visit(String &string)
 {
+  // Skip check for printf()'s format string (1st argument) and create the string
+  // with the original size. This is because format string is not part of bpf byte code.
+  if (func_ == "printf" && func_arg_idx_ == 0)
```

Comment:
Bit of a shame we cannot use `is_compiel_time_func` here. I've created #1539 to make that possible.